From 95ec37af68e9dd502daa23045416e38ce01161fd Mon Sep 17 00:00:00 2001
From: Xinyue Lu <i@7086.in>
Date: Tue, 27 Jan 2026 04:26:12 -0800
Subject: [PATCH 3/3] MMT/TLV: Skip unknown packet

---
 libavformat/mmttlv.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libavformat/mmttlv.c b/libavformat/mmttlv.c
index df9862fb1d..3226ff639b 100644
--- a/libavformat/mmttlv.c
+++ b/libavformat/mmttlv.c
@@ -287,8 +287,12 @@ static int mmttlv_read_packet(AVFormatContext *s, AVPacket *pkt)
     case HEADER_COMPRESSED_IP_PACKET:
         err = mmttlv_read_compressed_ip_packet(ctx, s, pkt, ctx->buf, size);
         break;
+    case TRANSMISSION_CONTROL_PACKET:
+        err = FFERROR_REDO;
+        break;
     default:
-        err = AVERROR_PATCHWELCOME;
+        av_log(s, AV_LOG_INFO, "Unsupported TLV packet type: 0x%02x\n", header[1]);
+        err = FFERROR_REDO;
     }
 
     if (err == FFERROR_REDO)
-- 
2.54.0

