From fea2a752e02f02e0fd821cf5971e4746d569fd47 Mon Sep 17 00:00:00 2001
From: huggy <i@huggy.moe>
Date: Tue, 21 Jul 2026 16:10:50 +0900
Subject: [PATCH] MMT/TLV: Skip invalid MMTP packets

---
 libavformat/mmttlv.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libavformat/mmttlv.c b/libavformat/mmttlv.c
index 3226ff639b..66263dce4b 100644
--- a/libavformat/mmttlv.c
+++ b/libavformat/mmttlv.c
@@ -149,6 +149,7 @@ static int mmttlv_read_compressed_ip_packet(
 #define PARTIAL_IPV6_HEADER_LENGTH (40 - 2)
 
     uint32_t       context_id;
+    int            err;
     struct Program *program;
 
     if (size < 3)
@@ -189,7 +190,10 @@ static int mmttlv_read_compressed_ip_packet(
         return AVERROR_INVALIDDATA;
     }
 
-    return ff_mmtp_parse_packet(program->mmtp, s, pkt, buf, size);
+    err = ff_mmtp_parse_packet(program->mmtp, s, pkt, buf, size);
+    if (err == AVERROR_INVALIDDATA)
+        return FFERROR_REDO;
+    return err;
 }
 
 static int mmttlv_read_packet(AVFormatContext *s, AVPacket *pkt)
-- 
2.54.0

